tracer/Directory.Build.props

```diff
@@ -3,7 +3,7 @@
     <IsPackable>false</IsPackable>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
-    <LangVersion>9.0</LangVersion>
+    <LangVersion>10.0</LangVersion>
```

Comment:
Any reason to not use `latest`?

Reply:
No reason 😄 done in https://github.com/DataDog/dd-trace-dotnet/pull/2076/commits/694cd0f45f832d4c9d4662ab3d4a1c2693f4fb99